### PR TITLE
feat: adding new preset (fastify-app)

### LINKED ISSUE #33: task to boostrap a fastify application
with routing, testing, ... inspiring from titus and fastify-cli

### PR DESCRIPTION
**Pull request by ramonmulia**
Preset to create a standard fastify application.

Closes #33

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,6 +16,10 @@ into one of the following:
 3. `Aws CDK` aws cdk project
     - it uses [aws-cdk-lib](https://aws.amazon.com/cdk/) to define AWS resources using code.
     - tests with [tap](https://github.com/tapjs/node-tap)
+4. `Fastify Application` it adds the following features:
+    - it uses [@vercel/ncc](https://github.com/vercel/ncc) to compile the application.
```
Comment (simoneb) [inline README.md:20]: why ncc for the fastify app?

File: README.md [hunk 2]
```diff
@@ -28,6 +32,7 @@ Where `$TASK_NAME` is one of the following:
 1. `github-action`.
 2. `fastify-plugin`
 3. `aws-cdk`
+4. `fastify-bootstrap`
```
Comment (simoneb) [inline README.md:35]: let's call it fastify-app

File: fastify-bootstrap/index.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+      'vscode'
+    ])
+    .save()
+  lines('.husky/pre-commit').add(['npm run build && npm run test']).save()
+
+  lines('README.md')
+    .set([`# ${appSlug}`, appDescription])
+    .save()
+
+  install(['husky', 'tap', 'standard', '@vercel/ncc'])
```
Comment (simoneb) [inline fastify-bootstrap/index.js:83]: we said that we wouldn't use standard, right?

File: fastify-bootstrap/static/.taprc [hunk 4]
```diff
@@ -0,0 +1,4 @@
+jobs: 1
+reporter: terse
+check-coverage: false
+timeout: 60
```
Comment (simoneb) [inline fastify-bootstrap/static/.taprc:4]: I'm not sure we need a custom tap configuration. Is this something that the fastify-cli creates?

File: fastify-bootstrap/static/test/routes/example.test.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+'use-strict'
+
+const { test } = require('tap')
+const { build } = require('../helper')
+
+test('example is loaded', async t => {
+  t.plan(1)
```
Comment (simoneb) [inline fastify-bootstrap/static/test/routes/example.test.js:7]: we don't need t.plan when using async tests and there are not callback based assertions

File: fastify-bootstrap/static/test/routes/root.test.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
+'use strict'
+
+const { test } = require('tap')
+const { build } = require('../helper')
+
+test('default root route', async t => {
+  t.plan(1)
```
Comment (simoneb) [inline fastify-bootstrap/static/test/routes/root.test.js:7]: remove

File: fastify-bootstrap/static/test/routes/root.test.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+  const app = await build(t)
+
+  const res = await app.inject({
+    url: '/'
+  })
+  t.same(JSON.parse(res.payload), { root: true })
```
Comment (simoneb) [inline fastify-bootstrap/static/test/routes/root.test.js:14]: I think there's a better way to get the json from the response. look it up, I think we're using it in the-fastify-workshop

**Review by simoneb — CHANGES_REQUESTED** [review]

**Review by simoneb — CHANGES_REQUESTED** [review]
overall looks good, see inline comments and the change we discussed about not using standard was not done

File: README.md (hunk 1 above)
Comment (ramonmulia) [reply README.md:20]: that's true e don't need that

File: fastify-bootstrap/static/.taprc (hunk 4 above)
Comment (ramonmulia) [reply fastify-bootstrap/static/.taprc:4]: No and we don't need this too.

Comment (simoneb) [reply fastify-bootstrap/static/.taprc]: I'm curious then, where did it come from?

Comment (ramonmulia) [reply fastify-bootstrap/static/.taprc]: It comes from it from the `fastify-cli` https://github.com/fastify/fastify-cli/blob/master/.taprc but I don't think we need that too

Comment (simoneb) [reply fastify-bootstrap/static/.taprc]: we don't need anything from fastify-cli itself, only the template for the fastify app, but not from the repo itself

Comment (ramonmulia) [reply fastify-bootstrap/static/.taprc]: that's true.

**Pull request merged**